src/circuit/gadget/ecc/chip/mul_fixed.rs

```diff
@@ -0,0 +1,529 @@
+        offset: usize,
+        scalar: &ScalarFixed,
+    ) -> Result<(), Error> {
+        // Copy the scalar decomposition (`k`-bit windows)
+        for (window_idx, window) in scalar.windows().iter().enumerate() {
+            copy(
+                region,
+                || format!("k[{:?}]", window),
+                self.window,
+                window_idx + offset,
+                window,
+                &self.perm,
+            )?;
+        }
+
+        Ok(())
+    }
+
+    fn initialize_accumulator(
+        &self,
+        region: &mut Region<'_, pallas::Base>,
+        offset: usize,
+        base: OrchardFixedBases,
+        scalar: &ScalarFixed,
+    ) -> Result<EccPoint, Error> {
+        // Recall that the message at each window `w` is represented as
+        // `m_w = [(k_w + 2) ⋅ 8^w]B`.
+        // When `w = 0`, we have `m_0 = [(k_0 + 2)]B`.
+        let m0 = {
+            let k0 = scalar.windows_field()[0];
+            let m0 = k0.map(|k0| base.generator() * (k0 + pallas::Scalar::from_u64(2)));
+            let m0 = m0.map(|m0| m0.to_affine().coordinates().unwrap());
+
+            let x = m0.map(|m0| *m0.x());
+            let x_cell = region.assign_advice(
+                || "m0_x",
+                self.x_p,
+                offset,
+                || x.ok_or(Error::SynthesisError),
+            )?;
+            let x = CellValue::new(x_cell, x);
+
+            let y = m0.map(|m0| *m0.y());
+            let y_cell = region.assign_advice(
+                || "m0_y",
+                self.y_p,
+                offset,
+                || y.ok_or(Error::SynthesisError),
+            )?;
+            let y = CellValue::new(y_cell, y);
+
+            EccPoint { x, y }
+        };
+
+        // Assign u = (y_p + z_w).sqrt() for `m0`
+        {
+            let k0 = scalar.windows_usize()[0];
+            let u0 = &base.u()[0];
+            let u0 = k0.map(|k0| u0.0[k0]);
+
+            region.assign_advice(|| "u", self.u, offset, || u0.ok_or(Error::SynthesisError))?;
+        }
+
+        // Copy `m0` into `x_qr`, `y_qr` cells on row 1 of the incomplete addition.
+        let x = copy(
+            region,
+            || "initialize acc x",
+            self.add_incomplete_config.x_qr,
+            offset + 1,
+            &m0.x,
+            &self.perm,
+        )?;
+        let y = copy(
+            region,
+            || "initialize acc y",
+            self.add_incomplete_config.y_qr,
+            offset + 1,
+            &m0.y,
+            &self.perm,
+        )?;
+
+        Ok(EccPoint { x, y })
+    }
+
+    fn add_incomplete(
+        &self,
+        region: &mut Region<'_, pallas::Base>,
+        offset: usize,
+        mut acc: EccPoint,
+        base: OrchardFixedBases,
+        scalar: &ScalarFixed,
+    ) -> Result<EccPoint, Error> {
+        // This is 2^w, where w is the window width
+        let h = pallas::Scalar::from_u64(constants::H as u64);
+
+        let base_value = base.generator();
```

Comment:
Likewise here:
![image](https://user-images.githubusercontent.com/4993799/122224228-d00a5680-ceab-11eb-8008-5e8f66e5c509.png)

This is because... (see next comment)